### DISCUSSION TITLE
statusline like widget?

### DISCUSSION THREAD

**Discussion by hemna** [ceccopierangiolieugenio/pyTermTk#258] [General]
Hey guys,
 I'm trying to figure out how to use pyTermTK to build an app that has a look like irssi in the terminal with a statusline at the top of the terminal, big center area with scrolling text and another status line at the bottom of the terminal minus 1 line, and the last line where the user enters text.

similar to this.
https://en.wikipedia.org/wiki/Irssi#/media/File:Irssi_1.2.3_screenshot.png

This seems like it should be simple, but I couldn't find anything like this in the widgets for status lines

**Comment by ceccopierangiolieugenio** [comment]
The reason I have not made a proper status bar widget is because I couldn't find a proper way to make it flexible,
someone require a status bar with function keys or drop down menus (mc), or info messages (vim), or anything powerline related.
pyTermTk offer a wide varety of widgets to achieve this and those can be ombined using the different layouts available (i.e. you can have a clock aligned on the right)
Anyway, I am always looking for suggestion, for example as I said I'd like to include a status/control bar widget if I can find the different usecases and figure out the easiest way to implement it.
I.e. I recently added the appTemplate widget because this is the most common layout used by the big amount of apps available.

About your request this is the most basic example that may achieve it.
It is just a showcase and you need to customize or add few more layers in order to achieve your results (I can help you with that).

https://gist.github.com/ceccopierangiolieugenio/45c618e91bfc4e55b9395bc094e4839d

[tryItOnline](https://ceccopierangiolieugenio.github.io/pyTermTk/sandbox/sandbox.html?fileUri=https://gist.githubusercontent.com/ceccopierangiolieugenio/45c618e91bfc4e55b9395bc094e4839d/raw/2f45cc048c9db3955aed8db4f401f611ef399deb/258-status-widget.py)

![image](https://github.com/ceccopierangiolieugenio/pyTermTk/assets/8876552/16944c8c-1e2f-450d-83d2-6bd6576d1725)

```pyhon
#!/usr/bin/env python3

import random
import TermTk as ttk

vertLayout = ttk.TTkVBoxLayout()
root = ttk.TTk(layout=vertLayout)

topStatus = ttk.TTkLabel(maxHeight=1)
bottomStatus = ttk.TTkLabel(maxHeight=1)
textArea = ttk.TTkTextEdit(lineNumber=True,readOnly=False)
prompt = ttk.TTkLineEdit(maxHeight=1)

statusStyle = {'default':  {'color': ttk.TTkColor.BG_BLUE}}
topStatus.setStyle(statusStyle)
bottomStatus.setStyle(statusStyle)

vertLayout.addWidgets([
                topStatus,
                textArea,
                bottomStatus,
                prompt,
            ])

topStatus.setText("Welcome To Arch Linux, Good Luck: https://archlinux.org | Rules: . . .")
bottomStatus.setText("[13:03] [Vulphere(+Ziw)] [2:liberachat/#archlinux(+Cnrt)]")
prompt.setText("[#archlinux]")
textArea.setText("12:47 Init"+("\n12:48 " + ("text "*random.randint(5,20)))*random.randint(50,200))

root.mainloop()
```
